project_admin/tests.py

```diff
@@ -1,3 +1,41 @@
-# from django.test import TestCase
+from django.test import TestCase
+import vcr
+import requests
+from django.urls import reverse
+from django.conf import settings
 
-# Create your tests here.
+my_vcr = vcr.VCR(path_transformer=vcr.VCR.ensure_suffix('.yaml'), cassette_library_dir='project_admin/cassettes')
+
+
+class LoginTest(TestCase):
+
+    def setUp(self):
+        settings.DEBUG = True
+        self.invalid_token = 'INVALID_TOKEN'
+        self.master_token = 'XitlFDXBqm5TRK8Vuh3Ey2cDFdiTWz7amKpot97H9Xfgak1qpvray0b0arQhvpEP'
+        self.project_info_url = 'https://www.openhumans.org/api/direct-sharing/project/?access_token={}'
+
+    @my_vcr.use_cassette()
+    def test_invalid_token(self):
+        request_url = self.project_info_url.format(self.invalid_token)
+        response = requests.get(request_url)
+        self.assertEqual(response.status_code, 401)
```

Comment:
@rosygupta I have a doubt on why would we require these tests, Usually tests are written to test the lines that we wrote or we could change. But here the `test_invalid_token` and `test_valid_token` are testing the API interface of openhumans which doesn't belong to this project (It belongs to https://github.com/OpenHumans/open-humans ), it's behaviour changes only when something in the OpenHumans main repository is changed.
So what could be a reason behind dropping these two tests in here in oh-proj-management? when these tests doesn't test anything related to oh-proj-management code?
Let me know if I am wrong in my understanding. Thank You.

Reply:
@eeshadutta In order to use this `oh-proj-management`, it is required to enter a token in. Because our app `oh-proj-management` does check token validity in order to decide on login success or failure, it makes sense to test that our communications between `oh-proj-management` and the main site (`open-humans`) work as expected for both token exchanges _and_ the actual login in our app (which also uses those tokens). Let me know if this clears things up at all, I would be happy elaborate!